### DISCUSSION TITLE
Why does q-container keep padding-top with 66 pixels?

### DISCUSSION THREAD

**Discussion by ivanjaros** [quasarframework/quasar#17454] [General - Components / Directives / etc] [answered]
![obrázok](https://github.com/user-attachments/assets/17bb364e-caa8-4086-b54b-ea4c40a682a1)

I tried to set height-hint for q-header to 82 which matches 50px of header height and 16+16 pixels of vertical margin but it does nothing. Where does this 66 come from?

This is my header:
```vue
<q-layout view="lhr lpr lfr">
    <q-header reveal :reveal-offset="100" class="bg-white text-deep-purple q-mx-lg q-my-md q-py-sm shadow-10">
      <q-toolbar class="justify-between">
        <div class="col-shrink">
          <q-toolbar-title>{{ title }}</q-toolbar-title>
        </div>
        <div class="col-shrink">
        ... some buttons
        </div>
      </q-toolbar>
    </q-header>
```

**Comment by ivanjaros** [comment]
It seems that the script that computes the offset for the q-container is not taking into consideration vertical margin of the header. Padding yes, but margin no. This is a *bug*.

**Reply by yusufkandemir** [reply]
As per the [alternative box model](https://developer.mozilla.org/en-US/docs/Learn/CSS/Building_blocks/The_box_model#the_alternative_css_box_model), which is the one through the Quasar code(_and pretty much any modern CSS framework_), padding is part of the element size. Margin is not part of the size in any box model.

**Comment by ivanjaros** [comment]
This is the issue. It only takes the height:

![obrázok](https://github.com/user-attachments/assets/3f20fa47-d517-418b-b442-53a120ecda65)

`quasar\src\components\page\QPageContainer.js`

**Comment by ivanjaros** [comment]
In order to fix the offset, I have to add `q-mt-xl` to the q-page in order cover the header's `q-my-md`, but that causes the page to develop scroll when there is not enough content to cause it.  So, again, **this is a bug**. I know no one reads this anyway, just like discord, but at least I can vent a bit.

**Reply by yusufkandemir** [reply]
If you don't alter the margin of the `q-header` and give the margin to the q-header instead, you will be fine. We are checking out and moderating Discord and GitHub Discussions as often as we can. I've probably seen your questions a lot of times as your avatar is very familiar. From what I remember, I would like to give you some friendly advice. You might want to spend more effort on your questions to clarify them and provide a reproduction. That would greatly improve the help rate, at least the attempts.

**Reply by ivanjaros** [reply]
> If you don't alter the margin of the q-header and give the margin to the q-header instead, you will be fine

something here does not add up.

also, the scroll is being caused by main getting min-height style assigned with invalid numbers. i can change the q-mt-xl on the q-page but then the main goes under the header and if i want to style the main the design will be broken visually. if i add top margin to the page container, it will screw up drawer. any hack will result in broken design. this should not happen in the first place if the computed top offset would count the margin of the header. there is no point in offsetting the page body from the header if margin is not included because then it causes this nonsensical "underlap".

**Reply by yusufkandemir** [reply]
Sorry, I meant apply it to q-toolbar instead of q-header

**Comment by ivanjaros** [comment, marked as answer]
I had to remove styling from header, added inside wrapper, apply styling to it. Then added top padding to q-page, so far seems to be working as desired. I guess the cause of this was the notion that one should treat q-layout children as merely sections and apply no styling to them whatsoever. They are more like slots rather than components.
